### DISCUSSION TITLE
minor realdebrid rclone update

### DISCUSSION THREAD

**Discussion by itsToggle** [itsToggle/plex_debrid#75] [Announcements]
Hey guys,

As you know, the realdebrid webdav implementation is still not very good - which is why Im still recommending to mount realdebrid using my fork. Some of the issues with the realdebrid webdav are:

- you cant delete files
- plex will often re-discover already downloaded content, for me its almost every time im refreshing the library
- during those refreshs, the realdebrid webdav re-generates a bunch of download links, so im getting hundrets of pages of download links. Plex will also re-detect intros for shows, which takes ages and consumes large amounts of bandwidth, every time the library is scanned.

For me, these issues dont happen on my rclone fork. 

Ive been working on a few minor updates to my rclone fork, which I wanted an opinion on before I publish them:

- Torrent file dates are more consistent, leading to no more rediscovering of already downloaded content
- Torrents are now automatically sorted in 3 subfolders: "shows", "movies" and "default".

The automatic sorting into subfolders works great, but if you have already setup a plex library with the current version of the rclone fork, plex will need to re-discover every movie and show (once). This will take some time and might annoy some/most of you, and I wanted to ask if I should implement these updates or not.

Are you guys interested in these updates? Or does the current fork work fine as is? What are your expiriences with the official realdebrid webdav?

**Comment by beljim** [comment]
Implement away in my opinion. I was wondering if folder seperation was possible as I've had a couple of things end up in the wrong category.  Or do the experental branch. These things need ti be tried and tested for the  script to evolve into the best version of itself.

**Reply by itsToggle** [reply]
very good point. the updated version is up on the rclone_rd github: https://github.com/itsToggle/rclone_RD

**Reply by beljim** [reply]
I'm sure it's not on your end but worth checking. I used to be able to delete files but since messing around with getting this installed on my Unraid server and adding the new rclone file I can't seem to delete anything. Not sure if it's something I did or not.
I'm getting
Dir.Remove failed to remove directory: directory not empty

I've set the write permissions and the mount is working no issues as read only.

**Reply by itsToggle** [reply]
Hm, it does look like this is on your end - i just checked and when I delete movies from within plex they are removed from realdebrid. I'm running rclone with the --daemon tag, so I'm not seeing any output, but have you checked wether or not files/torrents are deleted on realdebrid eventhough rclone prints the error? If the torrents really aren't deleted, I'll have a look at the code again.

When you use plex to delete a movie, plex will want to only remove the actual video file. My rclone fork however, will find the torrent that contains the video file, delete all corresponding unrestricted direct links to *all* files within the torrent from /downloads and then remove the torrent itself from /torrents. This might confuse the virtual filesystem a little and cause it to print this error. It should work non the less.

**Reply by beljim** [reply]
It was something my end after all. After last nights reboot it 's now deleting again :)

I'm curious what the cmount command does. I am using the regular 'rclone mount' as I didn't notice you have cmount in the instructions. 
Is cmount a typo or does it benefit the mount in some way. Couldn't find any info via Google.

**Reply by itsToggle** [reply]
that's good to hear! cmount is just the windows-equivalent of the mount command. I should make that clearer in the readme
